service/resource/kmskeyv1/delete.go

```diff
@@ -3,13 +3,76 @@ package kmskeyv1
 import (
 	"context"
 
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/kms"
+	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/operatorkit/framework"
 )
 
 func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteChange interface{}) error {
+	deleteAliasInput, err := toDeleteAliasInput(deleteChange)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	keyAlias := deleteAliasInput.AliasName
+	if keyAlias != nil && *keyAlias != "" {
+		if _, err := r.awsClients.KMS.DeleteAlias(&deleteAliasInput); err != nil {
+			return microerror.Mask(err)
+		}
+
+		key, err := r.awsClients.KMS.DescribeKey(&kms.DescribeKeyInput{
+			KeyId: keyAlias,
+		})
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		// AWS API doesn't allow to delete the KMS key immediately, but we can schedule its deletion
+		if _, err := r.awsClients.KMS.ScheduleKeyDeletion(&kms.ScheduleKeyDeletionInput{
+			KeyId:               key.KeyMetadata.KeyId,
+			PendingWindowInDays: aws.Int64(7),
```

Comment:
Minor but I'd like to have a constant for this.

Reply:
Sure, done